src/sagemaker/utils.py

```diff
@@ -58,6 +58,22 @@ def name_from_base(base, max_length=63, short=False):
     return '{}-{}'.format(trimmed_base, timestamp)
 
 
+def airflow_name_from_base(base):
+    """Append airflow execution_date macro to the provided string.
+
+    This function assures that the total length of the resulting string is not
+    longer than the specified max length, trimming the input parameter if necessary.
```

Comment:
We don't know the real length here. The macro '{{execution_date}}' will be evaluated in runtime. 

It might be possible that it will be evaluated to a time string with fixed length. Let me do more testing to confirm this. If it gets confirmed, I will change it in future PRs.

For now I would not make any change since taking more testing takes much time.